src/TauTriggerSFs2017.cc

```diff
@@ -56,38 +57,72 @@ int dmCheck(int dm )
 }
 
 
-TauTriggerSFs2017::TauTriggerSFs2017(const std::string& inputFileName, const std::string& trigger, const std::string& year, const std::string& tauWP, const std::string& wpType)
-  : inputFileName_(inputFileName),
-    trigger_(trigger),
+TauTriggerSFs2017::TauTriggerSFs2017(const std::string& trigger, const std::string& year, const std::string& tauWP, const std::string& wpType, const bool& emb_sfs)
+  : trigger_(trigger),
     year_(year),
     tauWP_(tauWP),
-    wpType_(wpType)
+    wpType_(wpType),
+    provide_emb_sfs_(emb_sfs)
 {
-  inputFile_ = new TFile(inputFileName_.data());
+  if (provide_emb_sfs_ && wpType_ != "DeepTau")
+  {
+      std::cerr << "Embedded scale factors are currently only provided for the DeepTau ID. ";
+      std::cerr << "You provided tau ID type: " << wpType_ << std::endl;
+      assert(0);
+  }
+  if (!provide_emb_sfs_ && wpType_ == "DeepTau")
+  {
+      std::cerr << "MC scale factors for the DeepTau ID are provided via the SFProvider interface. Aborting..." << std::endl;
+      assert(0);
+  }
+
+  std::string inputFileName = (boost::format("%s/src/TauAnalysisTools/TauTriggerSFs/data/tauTriggerEfficiencies%s.root") % std::getenv("CMSSW_BASE") % year).str();
```

Comment:
- Input file names must be provided in the constructor. Not all analysis frameworks are CMSSW-based, so CMSSW_BASE variable could be not set.
- Like this it should also be backwards compatible, right? (`emb_sfs = false` by default)

Reply:
* Does the first comment hold for the python interface as well? There it is implemented in the same way. If yes we would need to change this as well, right?
* Yes, the remark concerning the backwards compatibility only referred to the input file name as parameter of the constructor.

Reply:
Python interface can be used, for example, together with nanoAOD tools, which can be run standalone (without CMSSW). So, yes, better to change it as well. But you can keep back-compatibility: add file_name as an argument with default value None, and in case if it is not specified - load the default file name using CMSSW_BASE as it is currently done.

Reply:
I made the changes to the python interface as you proposed. For the C++ interface I overloaded the constructor with a version allows to construct the readout class without specifying the input file name and added back the previous constructor of the readout class.